File: src (VB.NET)/Z.ExtensionMethods.VB/Z.Data/System.Data.Common.DbCommand/DbCommand.ExecuteExpandoObject.vb

' Copyright (c) 2015 ZZZ Projects. All rights reserved
' Licensed under MIT License (MIT) (https://github.com/zzzprojects/Z.ExtensionMethods)
' Website: http://www.zzzprojects.com/
' Feedback / Feature Requests / Issues : http://zzzprojects.uservoice.com/forums/283927
' All ZZZ Projects products: Entity Framework Extensions / Bulk Operations / Extension Methods /Icon Library

Imports System.Data
Imports System.Data.Common

Public Module Extensions_649

	''' <summary>
	'''     A DbCommand extension method that executes the expando object operation.
	''' </summary>
	''' <param name="this">The @this to act on.</param>
	''' <returns>A dynamic.</returns>
    <System.Runtime.CompilerServices.Extension> _
    Public Function ExecuteExpandoObject(this As DbCommand) As Object
        Using reader As IDataReader = this.ExecuteReader()
            reader.Read()
            Return reader.ToExpandoObject()
        End Using
    End Function
End Module

File: src/Z.Data/System.Data.Common.DbCommand/DbCommand.ExecuteExpandoObject.cs
// Description: C# Extension Methods | Enhance the .NET Framework and .NET Core with over 1000 extension methods.
// Website & Documentation: https://csharp-extension.com/
// Issues: https://github.com/zzzprojects/Z.ExtensionMethods/issues
// License (MIT): https://github.com/zzzprojects/Z.ExtensionMethods/blob/master/LICENSE
// More projects: https://zzzprojects.com/
// Copyright © ZZZ Projects Inc. All rights reserved.
using System.Data;
using System.Data.Common;

public static partial class Extensions
{
    /// <summary>
    ///     A DbCommand extension method that executes the expando object operation.
    /// </summary>
    /// <param name="this">The @this to act on.</param>
    /// <returns>A dynamic.</returns>
    public static dynamic ExecuteExpandoObject(this DbCommand @this)
    {
        using (IDataReader reader = @this.ExecuteReader())
        {
            reader.Read();
            return reader.ToExpandoObject();
        }
    }
}